client/src/components/drop-down/drop-down.js

```diff
@@ -0,0 +1,32 @@
+import Component from '@/components/component';
+
+import styles from '@/styles/components/drop-down/drop-down.module.scss';
+
+export default class DropDown extends Component {
+  constructor(props) {
+    super(props);
+
+    this.$dom = this.createDom('div', {
+      className: styles['drop-down'],
+    });
+
+    this.render();
+    this.addEvent();
+  }
+
+  render = () => {
+    this.$dom.innerHTML = this._props.itemList
+      .map((item) => {
+        return `<div class="${styles['item']} ${styles[item.state]}">${item.label}</div>`;
+      })
+      .join('');
+  };
+
+  addEvent = () => {
+    this.$dom.querySelectorAll(`.${styles['item']}`).forEach(($item) => {
+      $item.addEventListener('click', () => {
+        this._props.onClick($item.innerHTML);
+      });
+    });
```

Comment:
```suggestion
    this.$dom.addEventListener('click', (e) => {
      if (e.target.classList.contains(styles.item)) {
        this._props.onClick(e.target.textContent);
      }
    });
```

event delegation을 쓰시는게 더 좋을 것 같습니다.

Reply:
event delegration 을 어떻게 적용해야할지 잘 생각이 안났었는데 잘 동작하게 만들어주셨네요 감사합니다.